VideoLocker/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -275,11 +276,75 @@ public void onFinish(RegisterResponse result) {
                         hideProgress();
 
                         if ( !result.isSuccess()) {
-                            String errorMessage = result.getValue();
-                            if(errorMessage == null || errorMessage.isEmpty()){
-                                errorMessage = getString(R.string.sign_up_error);
+
+                            // Response might contain error message for individual registration fields
+                            int errorCount = 0;
+
+                            for (IRegistrationFieldView fieldView : mFieldViews) {
+                                if (IRegistrationFieldView.FieldName.EMAIL
```

Comment:
And so if we make the error list a map, we can just iterate over the fields looking for any errors relevant to that field's name.